tools/src/main/java/org/apache/kafka/tools/VerifiableProducer.java

```diff
@@ -154,79 +153,45 @@ private static ArgumentParser argParser() {
-     * Since this method is not in Utils in the 0.8.X.X jars, we have to cheat a bit and duplicate.
-     */
-    public static Properties loadProps(String filename) throws IOException, FileNotFoundException {
-        Properties props = new Properties();
-        try (InputStream propStream = new FileInputStream(filename)) {
-            props.load(propStream);
-        }
-        return props;
-    }
-    
     /** Construct a VerifiableProducer object from command-line arguments. */
-    public static VerifiableProducer createFromArgs(String[] args) {
-        ArgumentParser parser = argParser();
-        VerifiableProducer producer = null;
-
-        try {
-            Namespace res;
-            res = parser.parseArgs(args);
-
-            int maxMessages = res.getInt("maxMessages");
-            String topic = res.getString("topic");
-            int throughput = res.getInt("throughput");
-            String configFile = res.getString("producer.config");
-            Integer valuePrefix = res.getInt("valuePrefix");
-
-            Properties producerProps = new Properties();
-            producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, res.getString("brokerList"));
-            producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
-                              "org.apache.kafka.common.serialization.StringSerializer");
-            producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
-                              "org.apache.kafka.common.serialization.StringSerializer");
-            producerProps.put(ProducerConfig.ACKS_CONFIG, Integer.toString(res.getInt("acks")));
-            // No producer retries
-            producerProps.put("retries", "0");
-            if (configFile != null) {
-                try {
-                    producerProps.putAll(loadProps(configFile));
-                } catch (IOException e) {
-                    throw new ArgumentParserException(e.getMessage(), parser);
-                }
-            }
-
-            producer = new VerifiableProducer(producerProps, topic, throughput, maxMessages, valuePrefix);
-        } catch (ArgumentParserException e) {
-            if (args.length == 0) {
-                parser.printHelp();
-                Exit.exit(0);
-            } else {
-                parser.handleError(e);
-                Exit.exit(1);
+    public static VerifiableProducer createFromArgs(ArgumentParser parser, String[] args) throws ArgumentParserException {
+        Namespace res = parser.parseArgs(args);
+
+        int maxMessages = res.getInt("maxMessages");
+        String topic = res.getString("topic");
+        int throughput = res.getInt("throughput");
+        String configFile = res.getString("producer.config");
+        Integer valuePrefix = res.getInt("valuePrefix");
+
+        Properties producerProps = new Properties();
+        if (configFile != null) {
+            try {
+                producerProps.putAll(Utils.loadProps(configFile));
```

Comment:
Changing to `Utils.loadProps` won't be safe for the reason noted in the comment that was deleted when the duplicated `loadProps` method was removed. Since we still have upgrade tests that run against 0.8 brokers, we still need the duplicated code.